velox/functions/prestosql/tests/ArithmeticTest.cpp

```diff
@@ -699,5 +699,153 @@ TEST_F(ArithmeticTest, truncate) {
   EXPECT_DOUBLE_EQ(truncate(123456789012345678901.23, -21).value(), 0.0);
 }
 
+TEST_F(ArithmeticTest, wilsonInterval) {
+  const auto wilsonIntervalUpper = [&](std::optional<int64_t> s,
+                                       std::optional<int64_t> n,
+                                       std::optional<double> z) {
+    return evaluateOnce<double>("wilson_interval_upper(c0,c1,c2)", s, n, z);
+  };
+  const auto wilsonIntervalLower = [&](std::optional<int64_t> s,
+                                       std::optional<int64_t> n,
+                                       std::optional<double> z) {
+    return evaluateOnce<double>("wilson_interval_lower(c0,c1,c2)", s, n, z);
+  };
+
+  // Verify that bounds checking is working.
+  VELOX_ASSERT_THROW(
+      wilsonIntervalUpper(-1, -1, -1),
+      "number of successes must not be negative");
+  VELOX_ASSERT_THROW(
+      wilsonIntervalUpper(0, -1, -1), "number of trials must be positive");
+  VELOX_ASSERT_THROW(
+      wilsonIntervalUpper(0, 0, -1), "number of trials must be positive");
+  VELOX_ASSERT_THROW(
+      wilsonIntervalUpper(0, 1, -1), "z-score must not be negative");
+  EXPECT_DOUBLE_EQ(wilsonIntervalUpper(0, 1, 0).value(), 0.0);
+  VELOX_ASSERT_THROW(
+      wilsonIntervalUpper(2, 1, 0),
+      "number of successes must not be larger than number of trials")
+
+  VELOX_ASSERT_THROW(
+      wilsonIntervalLower(-1, -1, -1),
+      "number of successes must not be negative");
+  VELOX_ASSERT_THROW(
+      wilsonIntervalLower(0, -1, -1), "number of trials must be positive");
+  VELOX_ASSERT_THROW(
+      wilsonIntervalLower(0, 0, -1), "number of trials must be positive");
+  VELOX_ASSERT_THROW(
+      wilsonIntervalLower(0, 1, -1), "z-score must not be negative");
+  EXPECT_DOUBLE_EQ(wilsonIntervalLower(0, 1, 0).value(), 0.0);
+  VELOX_ASSERT_THROW(
+      wilsonIntervalLower(2, 1, 0),
+      "number of successes must not be larger than number of trials")
+
+  // Verify correctness on simple inputs.
+  EXPECT_DOUBLE_EQ(wilsonIntervalUpper(3, 5, 0.5).value(), 0.7022485954964011);
+  EXPECT_DOUBLE_EQ(wilsonIntervalUpper(2, 10, 1).value(), 0.3509224591703222);
+  EXPECT_DOUBLE_EQ(wilsonIntervalUpper(7, 14, 1.6).value(), 0.6965892748731962);
+  EXPECT_DOUBLE_EQ(
+      wilsonIntervalUpper(1250, 1310, 1.96).value(), 0.9642524717143908);
+
+  EXPECT_DOUBLE_EQ(wilsonIntervalLower(3, 5, 0.5).value(), 0.48822759497978935);
+  EXPECT_DOUBLE_EQ(wilsonIntervalLower(2, 10, 1).value(), 0.10362299537513234);
+  EXPECT_DOUBLE_EQ(
+      wilsonIntervalLower(7, 14, 1.6).value(), 0.30341072512680384);
+  EXPECT_DOUBLE_EQ(
+      wilsonIntervalLower(1250, 1310, 1.96).value(), 0.9414883725395894);
+
+  // Verify correctness on extreme inputs.
+  EXPECT_DOUBLE_EQ(
+      wilsonIntervalUpper(9223372036854775807LL, 9223372036854775807LL, 1.6)
```

Comment:
Are there representations in std::numeric_limits for these values ? Those would be preferable.

Reply:
I've replaced the values with `std::numeric_limits<int64_t>::max()` and `(...)::max() / 10` where applicable. One concern I have is that this may make it harder to identify the actual values that are being passed to the function.